gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -202,10 +221,30 @@ def run(self) -> None:
         """Run the SceneOptimizer."""
         start_time = time.time()
 
-        # create dask client
-        cluster = LocalCluster(
-            n_workers=self.parsed_args.num_workers, threads_per_worker=self.parsed_args.threads_per_worker
-        )
+        # create dask cluster
+        if self.parsed_args.run_cluster:
+            workers = OmegaConf.load(
+                os.path.join(
+                    self.parsed_args.output_root, "gtsfm", "configs", self.parsed_args.cluster_config_name))["workers"]
+            scheduler = workers[0]
+            cluster = SSHCluster(
+                [scheduler] + workers,
+                scheduler_options={"dashboard_address": self.parsed_args.dashboard_port},
+                worker_options={
+                    "n_workers": self.parsed_args.num_workers,
+                    "nthreads": self.parsed_args.threads_per_worker,
+                },
+            )
+            client = Client(cluster)
+            # getting first worker's IP address and port to do IO
+            io_worker = list(client.scheduler_info()["workers"].keys())[0]
+            self.loader._input_worker = io_worker
+            self.scene_optimizer._output_worker = io_worker
+        else:
+            cluster = LocalCluster(
+                n_workers=self.parsed_args.num_workers, threads_per_worker=self.parsed_args.threads_per_worker
+            )
+            client = Client(cluster)
```

Comment:
do we not need to use the client and cluster variables anywhere?

Reply:
we use the cluster variable only to create the client, but we do not need to use the client variable

Reply:
weird, alright